domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt

```diff
@@ -721,8 +721,29 @@ class ProfileManagementController @Inject constructor(
     return imageFile.absolutePath
   }
 
-  private fun onlyLetters(name: String): Boolean {
-    return name.matches(Regex("^[ A-Za-z]+\$"))
+  private fun nameAllowed(name: String): Boolean {
+    return (
+      notEmptyNoSpacesAndContainsLetters(name) &&
+        noNumbers(name) &&
+        noSymbols(name) &&
+        noRepeatedUseOfAllowedSymbols(name)
+      )
+  }
+
+  private fun notEmptyNoSpacesAndContainsLetters(name: String): Boolean {
+    return name.matches(Regex("^.[\\w\\u00BF-\\u1FFF\\u2C00-\\uD7FF.'\\-]+\$"))
```

Comment:
Similarly to below, you could have a separate check that uses ``isWhitespace`` to simplify this just to checking for letters. That being said, would ``isLetter`` also work instead of using the pattern?

Reply:
This method ensures that the name isn't empty (doesn't allow blank space, next line), but at the same time it matches letters of "any language" -> so isLetters() can't be used. Please Note that this regex pattern doesn't reject any symbols, it merely returns `true` when it there is a `letter` (of any language) is present in the name.

Reply:
Why wouldn't ``isLetter`` work? It's my understanding that it should be checking for letters in roughly the same range that you're using here.

Also, ``isWhitespace`` should be the same as ``\\w`` (and you could specifically verify this by adding tests to cover various whitespace).

Reply:
Reminder to respond to this--isLetter would be a large simplification.